src/server/project.ts

```diff
@@ -803,6 +803,22 @@ namespace ts.server {
             return result;
         }
 
+        /* @internal */
+        getFileNamesWithRedirectInfo(includeProjectReferenceRedirectInfo: boolean) {
+            const fileNames = this.getFileNames();
+            if (includeProjectReferenceRedirectInfo) {
+                return fileNames.map((fileName): protocol.FileWithProjectReferenceRedirectInfo => ({
+                    fileName,
+                    isSourceOfProjectReferenceRedirect: this.isSourceOfProjectReferenceRedirect(fileName)

```

Comment:
In if and else both you are returning FileWithProjectReferenceRedirectInfo why not write this function as 
```ts
getFileNamesWithRedirectInfo(includeProjectReferenceRedirectInfo: boolean) {
          return  this.getFileNames().map((fileName): protocol.FileWithProjectReferenceRedirectInfo => ({
                    fileName,
                    isSourceOfProjectReferenceRedirect: includeProjectReferenceRedirectInfo && this.isSourceOfProjectReferenceRedirect(fileName)
          }));
}
```

Reply:
Fixed - I had done it this way to minimize accesses of includeProjectReferenceRedirectInfo, but this is more readable.